zookeeper-server/src/test/java/org/apache/zookeeper/ZKTestCase.java

```diff
@@ -75,5 +77,30 @@ public void failed(Throwable e, Description method) {
         }
 
     };
+    public interface WaitForCondition {
+        /**
+         * @return true when success
+         */
+        boolean evaluate();
+    }
+
+    /**
+     * Wait for condition to be true; otherwise fail the test if it exceed
+     * timeout
+     * @param msg       error message to print when fail
+     * @param condition condition to evaluate
+     * @param timeout   timeout in seconds
+     * @throws InterruptedException
+     */
+    public void waitFor(String msg, WaitForCondition condition, int timeout) throws InterruptedException {
```

Comment:
this is a very useful helper function, great that we will have it also for the 3.5 branch!